src/treeview.ts

```diff
@@ -0,0 +1,186 @@
+import * as path from "path";
+import * as fs from "fs";
+import { LanguageClient, Disposable } from "vscode-languageclient";
+import {
+  TreeDataProvider,
+  TreeItem,
+  Event,
+  EventEmitter,
+  TreeItemCollapsibleState,
+  window,
+  OutputChannel,
+  Uri,
+  TreeView,
+  ExtensionContext
+} from "vscode";
+import {
+  TreeViewNode,
+  MetalsTreeViews,
+  MetalsTreeViewChildren,
+  MetalsTreeViewDidChange,
+  MetalsTreeViewVisibilityDidChange,
+  MetalsRevealTreeView,
+  MetalsTreeViewParent
+} from "./protocol";
+
+export function startTreeView(
+  client: LanguageClient,
+  out: OutputChannel,
+  context: ExtensionContext,
+  packageJson: any
+): MetalsTreeViews {
+  let allProviders: Map<string, MetalsTreeDataProvider> = new Map();
+  let allViews: Map<string, TreeView<string>> = new Map();
+  let viewIds: string[] = packageJson.contributes.views["metals-explorer"].map(
+    (view: { id: string }) => view.id
+  ) as string[];
```

Comment:
this cast is redundant (and probably also the type annotation)
```suggestion
  );
```

Reply:
Right, makes sense. A bit shady that ascribing the parameter type in the lambda casts the type 😨